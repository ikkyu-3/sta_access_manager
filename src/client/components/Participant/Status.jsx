// @flow
import type { ParticipantStatusProps } from "@types";

import * as React from "react";
import styles from "./style.css";

function Status(props: ParticipantStatusProps): React.Node {
  const { isEntry } = props;
  const { columnStatus, entry, exit } = styles;

  const className = isEntry
    ? `${columnStatus} ${entry}`
    : `${columnStatus} ${exit}`;
  const value = isEntry ? "入" : "退";

  return <div className={className}>{value}</div>;
}
export default Status;
