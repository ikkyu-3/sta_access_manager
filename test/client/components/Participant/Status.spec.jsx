/* eslint-disable react/jsx-boolean-value */
import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Status from "@components/Participant/Status";

describe("Participant.Status.jsxのテスト", () => {
  describe("スナップショット", () => {
    it("正しいレンダリング", () => {
      const tree = renderer.create(<Status isEntry />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("コンポーネントのテスト", () => {
    it("isEntryがtrueの時、「入」と表示される", () => {
      const status = shallow(<Status isEntry={true} />);
      expect(status.hasClass("entry")).toBeTruthy();
      expect(status.text("入")).toBeTruthy();
    });

    it("isEntryがfalseの時、「退」と表示される", () => {
      const status = shallow(<Status isEntry={false} />);
      expect(status.hasClass("exit")).toBeTruthy();
      expect(status.text("退")).toBeTruthy();
    });
  });
});
