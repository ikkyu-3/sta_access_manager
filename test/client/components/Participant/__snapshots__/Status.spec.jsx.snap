// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant.Status.jsxのテスト スナップショット 正しいレンダリング 1`] = `
<div
  className="columnStatus entry"
>
  入
</div>
`;
